ProfessorAssessor/View Controllers/ComposeViewController.m

```diff
@@ -1,13 +1,185 @@
 #import "ComposeViewController.h"
+#import "ProfessorSelectionViewController.h"
+#import "CourseSelectionViewController.h"
+#import "Networker.h"
+#import "User.h"
 
-@interface ComposeViewController ()
+static NSNumberFormatter *numberFormatter = nil;
+
+@interface ComposeViewController () <ProfessorSelectionViewControllerDelegate, CourseSelectionViewControllerDelegate, UITextViewDelegate>
+
+@property (nonatomic, strong) IBOutlet UILabel *professorLabel;
+@property (nonatomic, strong) IBOutlet UILabel *courseLabel;
+@property (nonatomic, strong) IBOutlet UILabel *ratingLabel;
+@property (nonatomic, strong) IBOutlet UILabel *reviewLabel;
+@property (nonatomic, strong) IBOutlet UILabel *characterCountLabel;
+@property (nonatomic, strong) IBOutlet UIButton *chooseCourseButton;
+@property (nonatomic, strong) IBOutlet UIBarButtonItem *submitButton;
+
+- (IBAction)ratingViewChanged:(HCSStarRatingView *)sender;
+- (IBAction)ratingFieldChanged:(UITextField *)sender;
 
 @end
 
 @implementation ComposeViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+    [self decorateContentView];
+
+    if (self.professor) {
+        self.professorName.text = self.professor.name;
+    } else {
+        [self viewsEnabled:NO];
+    }
+}
+
+- (IBAction)submitReview:(UIBarButtonItem *)sender {
+    if (self.course == nil) {
+        [self presentAlertWithTitle:nil withMessage:@"Please select a course"];
+    } else if (self.ratingField.text.length == 0) {
+        [self presentAlertWithTitle:nil withMessage:@"Please enter a rating"];
+    } else if (self.content.text.length == 0) {
+        [self presentAlertWithTitle:nil withMessage:@"Please type your review of this professor"];
+    } else {
+        NSNumber *ratingValue = [NSNumber numberWithDouble:[self.ratingField.text doubleValue]];
+
+        [Networker
```

Comment:
I think this is missing some element of user feedback. A spinner, overlay, disabled buttons, something to let them know that work is happening. The completion should undo whatever this is to show that they can interact again.